src/Analyzers/CSharp/Analyzers/UseParameterNullChecking/CSharpUseParameterNullCheckingDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,242 @@
+            if (!option.Value)
+            {
+                return;
+            }
+
+            var node = context.Node;
+            var block = node switch
+            {
+                MethodDeclarationSyntax methodDecl => methodDecl.Body,
+                LocalFunctionStatementSyntax localFunctionStatement => localFunctionStatement.Body,
+                _ => throw ExceptionUtilities.UnexpectedValue(node)
+            };
+            if (block is null)
+            {
+                return;
+            }
+
+            var methodSymbol = semanticModel.GetDeclaredSymbol(node, cancellationToken);
+            if (methodSymbol is null)
+            {
+                return;
+            }
+
+            foreach (var statement in block.Statements)
+            {
+                switch (statement)
+                {
+                    // if (param == null) { throw new ArgumentNullException(nameof(param)); }
+                    // if (param is null) { throw new ArgumentNullException(nameof(param)); }
+                    // if (object.ReferenceEquals(param, null)) { throw new ArgumentNullException(nameof(param)); }
+                    case IfStatementSyntax ifStatement:
+                        ExpressionSyntax left, right;
+                        switch (ifStatement)
+                        {
+                            case { Condition: BinaryExpressionSyntax { OperatorToken.RawKind: (int)SyntaxKind.EqualsEqualsToken } binary }:
+                                left = binary.Left;
+                                right = binary.Right;
+                                break;
+                            case { Condition: IsPatternExpressionSyntax { Expression: var patternInput, Pattern: ConstantPatternSyntax { Expression: var patternExpression } } }:
+                                left = patternInput;
+                                right = patternExpression;
+                                break;
+                            case { Condition: InvocationExpressionSyntax { Expression: var receiver, ArgumentList.Arguments: { Count: 2 } arguments } }:
+                                if (!referenceEqualsMethod.Equals(semanticModel.GetSymbolInfo(receiver, cancellationToken).Symbol))
+                                {
+                                    continue;
+                                }
+
+                                left = arguments[0].Expression;
+                                right = arguments[1].Expression;
+                                break;
+
+                            default:
+                                continue;
+                        }
+
+                        IParameterSymbol? parameter;
+                        if (!areOperandsApplicable(left, right, out parameter)
+                            && !areOperandsApplicable(right, left, out parameter))
```

Comment:
```suggestion
                        if (!AreOperandsApplicable(left, right, out var parameter)
                            && !AreOperandsApplicable(right, left, out parameter))
```

though this comes down to taste.  leave if you prefer as is (for the parameter decl).

Reply:
Code has changed for unrelated reasons, so resolving the thread.